lib/util.js

```diff
@@ -616,8 +616,18 @@ exports.log = function() {
  * @param {function} ctor Constructor function which needs to inherit the
  *     prototype.
  * @param {function} superCtor Constructor function to inherit prototype from.
+ * @throws {TypeError} Will error if either constructor is null, or if
+ *     the super constructor lacks a prototype.
  */
 exports.inherits = function(ctor, superCtor) {
+
+  if (isNullOrUndefined(ctor))
```

Comment:
In the codebase, the `util.isXXX()` functions have been phased out since [`6ac8bdc`](https://github.com/iojs/io.js/commit/6ac8bdc0aba5f60f4b4f2da5abd36d664062aa40), could you change them here and below to an inline alternative?